cli/templatepush.go

```diff
@@ -164,24 +166,33 @@ func (r *RootCmd) templatePush() *clibase.Cmd {
 				return err
 			}
 
-			template, err := client.TemplateByName(inv.Context(), organization.ID, name)
+			resp, err := uploadFlags.upload(inv, client)
 			if err != nil {
 				return err
 			}
 
-			err = uploadFlags.checkForLockfile(inv)
-			if err != nil {
-				return xerrors.Errorf("check for lockfile: %w", err)
-			}
-
-			resp, err := uploadFlags.upload(inv, client)
+			tags, err := ParseProvisionerTags(provisionerTags)
 			if err != nil {
 				return err
 			}
 
-			tags, err := ParseProvisionerTags(provisionerTags)
+			template, err := client.TemplateByName(inv.Context(), organization.ID, name)
 			if err != nil {
```

Comment:
Need to check if the error is due to "not found":

```
if sdkErr, ok := codersdk.AsError(err); ok && sdkErr.StatusCode() !=  http.StatusNotFound {
	return xerrors.Errorf("get template: %w", err)
}
// otherwise, template is not found, ask if user wants to create
```